pre_commit/repository.py

```diff
@@ -242,6 +247,55 @@ def _venvs(self):
         return tuple(ret)
 
 
+class MetaRepository(LocalRepository):
+    # Note: the hook `entry` is passed through `shlex.split()` by the command
+    # runner, so to prevent issues with spaces and backslashes (on Windows) it
+    # must be quoted here.
+    meta_hooks = {
+        'check-useless-excludes': {
+            'name': 'Check for useless excludes',
+            'files': '.pre-commit-config.yaml',
+            'language': 'system',
+            'entry': pipes.quote(sys.executable),
+            'args': ['-m', 'pre_commit.meta_hooks.check_useless_excludes'],
```

Comment:
one benefit of `entry` is it gets `shlex`'d for us.  so we can do this:

```python
'entry': '{} -m {}'.format(pipes.quote(sys.executable), check_useless_excludes.__name__),
```

instead of (error prone) hardcoding the module names, we can import them and use `__name__` as well

Reply:
+1 for using the module's `__name__`, I didn't think of that.

Reply:
Annoyingly, this lead to a cyclic import if, for example, the hook tries to import `Repository` or `Runner`.

Reply:
hmmmm womp.

Perhaps make a

```python
@cached_property
def manifest_hooks(self):
    from pre_commit.meta_hooks import check_useless_excludes
    ...
```

to avoid the circular imports? _(circular imports are such a bummer)_